GitUI/Infrastructure/Telemetry/ThemingTelemetryInitializer.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Drawing;
+using GitExtUtils.GitUI.Theming;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.Extensibility;
+
+namespace GitUI.Infrastructure.Telemetry
+{
+    internal class ThemingTelemetryInitializer : ITelemetryInitializer
+    {
+        public void Initialize(ITelemetry telemetry)
+        {
+            var properties = telemetry.Context.GlobalProperties;
+            ThemeId themeId = Theming.ThemeModule.Settings.Theme.Id;
+            properties["Theme name"] = themeId.Name;
+            properties["Theme is builtin"] = themeId.IsBuiltin.ToString();
+            properties["Theme use system visual style"] = Theming.ThemeModule.Settings.UseSystemVisualStyle.ToString();
+
+            if (Theming.ThemeModule.Settings.Theme != Theme.Default)
+            {
+                foreach ((AppColor appColor, Color color) in Theming.ThemeModule.Settings.Theme.AppColorValues)
```

Comment:
Do all colors need to be tracked?
This could be identifying users?
Is the dark theme included?

Reply:
Agree with Gerhard, I don't think it is necessary to capture such detailed information.
The biggest issue with this approach I see is telemetry payload bloat.

Reply:
ok, removed tracking individual colors